include/boost/geometry/algorithms/detail/buffer/turn_in_piece_visitor.hpp

```diff
@@ -77,6 +79,44 @@ class turn_in_piece_visitor
         return false;
     }
 
+    // Returns true if the turn is part of a cluster, and one of the other turns in the same
+    // cluster is involved in the same piece as the operations in the turn.
+    template <typename Turn, typename Piece>
+    inline bool skip_by_same_cluster(Turn const& turn, Piece const& piece) const
+    {
+        if (turn.cluster_id <= 0)
```

Comment:
what does a negative cluster_id means and why we are not searching for it? It is the case that m_clusters does not contain negative values so this condition speeds up the search?

Reply:
👍 I removed it, the condition was indeed redundant and wrong. Thanks.